backend/analytics_server/mhq/exapi/github.py

```
@@ -271,3 +279,229 @@ def _fetch_workflow_runs(page: int = 1):
+            if response.status_code != HTTPStatus.OK:
+                raise GithubException(
+                    response.status_code,
+                    f"Failed to fetch timeline events: {response.text}",
+                )
+
+            try:
+                return response.json()
+            except ValueError as e:
+                raise GithubException(
+                    HTTPStatus.INTERNAL_SERVER_ERROR, f"Invalid JSON response: {str(e)}"
+                )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)